sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -0,0 +1,523 @@
+    claimInfo.setClaiminfo(claimInfoHTLCSerialized);
+    return claimInfo.serializeBinary().toString();
+}
+
+// Create a SHA-256 hash over an ASCII string
+function createSHA256HashBase64(preimage)
+{
+    return crypto.createHash('sha256').update(preimage).digest('base64');
+}
+
+
+/**
+ * First/second step of a Hashed Time Lock Contract
+ * - Lock a unique asset instance using a hash
+ **/
+const createHTLC = async (
+    contract: Contract,
+    assetType: string,
+    assetID: string,
+    recipientECert: string,
+    hashValue: string,
+    expiryTimeSecs: number,
+): Promise<{ preimage: any; result: any }> => {
+
+    if (!contract)
+    {
+        logger.error("Contract handle not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!assetType)
+    {
+        logger.error("Asset type not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!assetID)
+    {
+        logger.error("Asset ID not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!recipientECert)
+    {
+        logger.error("Recipient ECert not supplied");
+        return { preimage: "", result: false };
+    }
+
+    let defaultExpiryTimeSecs = 0, preimage = "";
+    if (hashValue && hashValue.length > 0)
+    {
+        defaultExpiryTimeSecs = 5 * 60     // 5 mins: this is the 't' used to time out the second contract of the HTLC pair
+    }
+    else
+    {
+        defaultExpiryTimeSecs = 10 * 60     // 10 mins: this is the '2t' used to time out the first contract of the HTLC pair
+
+        // Create a random preimage of 20 bytes
+        for (let i = 0 ; i < 20 ; i++)
+        {
+            preimage += String.fromCharCode(Math.floor(Math.random() * 256))
+        }
+
+        // Hash the preimage
+        hashValue = createSHA256HashBase64(preimage);
+    }
+    const currTimeSecs = Math.floor(Date.now()/1000);   // Convert epoch milliseconds to seconds
+    if (expiryTimeSecs <= currTimeSecs)
+    {
+        logger.warn("Supplied HTLC expiry time is invalid or in the past: %d", expiryTimeSecs);
+        expiryTimeSecs = currTimeSecs + defaultExpiryTimeSecs;
+    }
+
+    const assetExchangeAgreementStr = createAssetExchangeAgreementSerialized(assetType, assetID, recipientECert, "");
+    const lockInfoStr = createAssetLockInfoSerialized(hashValue, expiryTimeSecs);
+
+    // Normal invoke function
+    const [result, submitError] = await helpers.handlePromise(
+        contract.submitTransaction("LockAsset", assetExchangeAgreementStr, lockInfoStr),
+    );
+    if (submitError) {
+        throw new Error(`LockAsset submitTransaction Error: ${submitError}`);
+    }
+    return { preimage: preimage, result: result };
+};
+
+/**
+ * First/second step of a Hashed Time Lock Contract
+ * - Lock a set of fungible assets using a hash
+ **/
+const createFungibleHTLC = async (
+    contract: Contract,
+    assetType: string,
+    numUnits: number,
```

Comment:
I think this loss in precision could be a problem, and one I suspect we would need to address at some point (hopefully without breaking compatibility). There is an open PR to address this issue by enabling protobuf to use `BigInt` instead of `number` for 64bit ints https://github.com/protobufjs/protobuf.js/pull/1557
We can defer this for now.